clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -617,7 +618,7 @@ public boolean commitOffsetsSync(Map<TopicPartition, OffsetAndMetadata> offsets,
 
             now = time.milliseconds();
             remainingMs = timeoutMs - (now - startMs);
-        } while (remainingMs > 0);
+        } while (remainingMs > 0 && !isFutureCompleted);
```

Comment:
OK, I filed https://issues.apache.org/jira/browse/KAFKA-5606 to track this discussion separately from this PR.